internal/metrics/metrics.go

```diff
@@ -57,7 +57,7 @@ var (
 			Name:      "upstream_rpc_requests_total",
 			Help:      "Count of total RPC requests forwarded to upstreams.",
 		},
-		[]string{"endpoint_id", "url"},
+		[]string{"client", "endpoint_id", "url", "jsonrpc_method"},
```

Comment:
Minor: is there a way to more generically add client to all metrics that are client-specific?

Reply:
It seems like the easiest way to do that is create our own interface/methods to add metrics. I think something that conveniently includes `context` would be nice, like:

```
		recordLatencyWithLabels(ctx, id, configToRoute.HTTPURL, requestBody.Method, HTTPReponseCode, JSONRPCResponseCode,
```

Or maybe we can stuff all of this other info into context and the method will look like:

```
recordLatency(ctx)
```

And the method figures out what labels to add based on what's in the context

Reply:
Definitely some improvements we can make can add for recording metrics. Will keep it in mind on the next PR

Reply:
>  create our own interface/methods to add metrics
If we do this, we're still defining individual functions and promauto metrics that manually define the `client` tag and  pull it out of the context.

Nbd! Was curious if there was some easy way to add default tags to each metric.